tutorials/js/data/events.js

```diff
@@ -0,0 +1,60 @@
+    },
+    {
+      name: 'Document Bindings',
+      index: 4,
+      // template_highlights: [{start: {line: 0, ch: 6}, end: {line: 0, ch: 12}}],
+      description_html: '<p>Adds an event binding to the document with delegation still working as expected</p><ul><li>Bindable by prepending <code>doc-</code> to any event type</li></ul><p>In the example below, there are four events being listened to. <code>mousedown</code> and <code>mouseenter</code> are regular events, and <code>doc-mouseup</code> and <code>doc-mouseleave</code> are delegated to the document.</p><p>Test each of these events below and notice the delegate in the status line below the event area.</p>',
+      template: '<p>Test events in the colored area below.</p>\n<div class="js-event-area pure-u-1" style="height:300px; background-color:#71A3AF;"></div>\n<div>captured: {{message}}</div>',
+      js: "var AppView = View.extend({\n  events: {\n    'mousedown .js-event-area': 'mousedownEvent',\n    'doc-mouseup .js-event-area': 'mouseupDelegated',\n    'mouseenter .js-event-area': 'mouseenterEvent',\n    'doc-mouseleave .js-event-area': 'mouseleaveDelegated'\n  },\n  mousedownEvent: function(e) {\n    this.setEventMessage('mousedown | delegate: ' + e.delegateTarget.nodeName);\n  },\n  mouseupDelegated: function(e) {\n    this.setEventMessage('mouseup | delegate: ' + e.delegateTarget.nodeName);\n  },\n  mouseenterEvent: function(e) {\n    this.setEventMessage('mouseenter | delegate: ' + e.delegateTarget.nodeName);\n  },\n  mouseleaveDelegated: function(e) {\n    this.setEventMessage('mouseleave | delegate: ' + e.delegateTarget.nodeName);\n  },\n  setEventMessage: function(msg) {\n    var self = this;\n    this.model.set('message', msg);\n    clearTimeout(this.messageTimeout);\n    this.messageTimeout = setTimeout(function() {\n      self.model.set('message', '---');\n    }, 1200);\n  }\n});\n\nnew AppView({\n  el: '#app',\n  template: compiledTemplates.app_view,\n  model: new AppModel(),\n  dynamicInitialize: true\n});"
```

Comment:
AppModel is used without being defined. Looks like app.js leaks variables into this scope accidentally, which is why View/Model/Collection are usable. If you look at TodoMVC, it does: ```var View = tungsten.backbone.View```